### PR TITLE
Template field removal migration

### PR DESCRIPTION
**Pull request by jsangmeister**

### CODE REVIEW

File: openslides_backend/migrations/migrations/0042_remove_template_fields.py [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+                FieldStrategy.Merge,
+                FieldStrategy.MergeToJSON,
+                FieldStrategy.MoveToMeetingUser,
+                FieldStrategy.MoveToMeetingUserAndReplace,
+            ):
+                name_func = lambda field: field.replace("_$", "")  # noqa: E731
+            elif resolved_strategy is FieldStrategy.ReplaceWithMeetingUsers:
+                name_func = lambda field: f"meeting_{field}"  # noqa: E731
+            else:
+                name_func = lambda field: field  # noqa: E731
```
Comment (r-peschke) [inline openslides_backend/migrations/migrations/0042_remove_template_fields.py:383]: Also here, raise an exception. This only occurs by a template field without a strategy

File: tests/system/migrations/test_0042_remove_template_fields.py [hunk 2]
```diff
@@ -0,0 +1,1320 @@
...
+    )
+    write(
+        {
+            "type": "delete",
+            "fqid": "meeting/43",
+        }
+    )
+    finalize("0042_remove_template_fields")
+
+    assert_model(
```
Comment (r-peschke) [inline tests/system/migrations/test_0042_remove_template_fields.py:756]: Please assert also the deleted meeting/43

File: openslides_backend/migrations/migrations/0042_remove_template_fields.py [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+                name_func = lambda _: strategy["name"]  # noqa: E731
+            elif isinstance(strategy["name"], dict):
+                name_map = strategy["name"]
+                name_func = (  # noqa: E731
+                    lambda field: name_map[field]
+                    if field in name_map
+                    else field.replace("$", "")
+                )
+            else:
+                name_func = strategy["name"]
```
[the remark this answers is not in the document]
Comment (jsangmeister) [reply openslides_backend/migrations/migrations/0042_remove_template_fields.py:368]: This and the other were fragments from earlier versions of the migration, where a e.g. lambda function was used directly. I removed it now

File: tests/system/migrations/test_0042_remove_template_fields.py (hunk 2 above)
Comment (jsangmeister) [reply tests/system/migrations/test_0042_remove_template_fields.py:756]: done

**Comment by jsangmeister** [comment]
I also refactored the `resolve_strategy` method a bit to get rid of the `noqa`s - I finally had a good idea how to do that

**Review by r-peschke — APPROVED** [review]

**Pull request merged**